account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -84,24 +77,149 @@
 			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
 		}
 
-		UserBean userBean = userMapper.userWSToBean(userWS);
+		UserBean userBean = userMapper.toBean(userWS);
 
 		UserBean user = accountService.createUser(userBean);
 
 		try {
-			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.REGISTRATION.name());
+			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.ACTIVATION.name());
+		} catch (MessagingException e) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.EMAIL_MESSAGING_EXCEPTION.name(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		return new ResponseEntity<>(userMapper.toWS(user), HttpStatus.CREATED);
+	}
+
+	@GetMapping("/activate/{token}")
+	public ResponseEntity<?> activateAccount(@PathVariable("token") String token) {
+		Optional<UserBean> user = accountService.findByToken(token);
+
+		if (!user.isPresent()) {
+			return new ResponseEntity<Object>(new ApiErrorsWS(ErrorCode.INVALID_TOKEN.name(), "Invalid Token."), HttpStatus.BAD_REQUEST);
+		}
+		accountService.activateUser(user.get().getId());
+
+		return new ResponseEntity<>(userMapper.toWS(user.get()), HttpStatus.OK);
+	}
+
+	@PostMapping("/issue-token")
+	@RequiresAuthentication
+	public ResponseEntity<?> issueToken(@Validated @RequestBody UserWS userWS) {
```

Comment:
Can you try this 
```	
	@PostMapping("/issue-token")
	@RequiresAuthentication
	public ResponseEntity<?> issueToken() {
		String uuid = (String) SecurityUtils.getSubject().getPrincipal();
		UserBean user = accountService.findByUuid(uuid).orElse(null);
		JWTAuthenticationToken jwtToken = jwtService.createJwtToken(user);

		log.debug("Success create jwt token. jwtToken={}", jwtToken.getToken());

		return new ResponseEntity<>(jwtService.createJwtHeader(jwtToken.getToken()), HttpStatus.OK);
	}
```
You already set username/password on header. no need to pass in again from request body.

Reply:
this works too. Fixed.